app/models/endpoint.rb

```diff
@@ -11,7 +11,13 @@ class Endpoint < ApplicationRecord
   attribute :availability_status, :string
   validates :availability_status, :inclusion => { :in => %w[available unavailable] }, :allow_nil => true
 
+  before_save :set_default, :if => proc { source.endpoints.count.zero? }
+
   def base_url_path
     URI::Generic.build(:scheme => scheme, :host => host, :port => port, :path => path).to_s
   end
+
+  def set_default
+    self.default = true
```

Comment:
If they do send in `self.default = false` would this over write that?

Reply:
Correct - this will make sure that the first endpoint added to a source will be marked as default which seems like a pretty common use-case when creating source/endpoint/authentication via the api.

Reply:
This really looks fine, but having a test proving your `EDIT` comment would be great. Other than that this is ready to go.

Reply:
@syncrou good idea, added a spec that makes sure a delete/re-add of endpoint marks it as default again.